src/java.base/share/classes/jdk/internal/reflect/ReflectionFactory.java

```diff
@@ -430,6 +433,69 @@ private final MethodHandle findReadWriteObjectForSerialization(Class<?> cl,
         }
     }
 
+    public final MethodHandle defaultReadObjectForSerialization(Class<?> cl) {
+        if (! isValidSerializable(cl)) {
+            return null;
+        }
+
+        return SerializationBytecodeGenerator.defaultReadObjectForSerialization(cl);
```

Comment:
My hope is that the method handle would be more easily inlined when each one is a separate constant. I'd feel less confident that this would be the case if I indirected through `List`. I think it would also increase the size of the generated method as well, though I'm not sure if there actually is any practical consequence to this.

Also, this was easier. ;-)

Reply:
Oh, this comment landed on the wrong spot for some reason. To respond to the actual post here, I figured that the serialization library is likely to perform this caching anyway, so adding another layer here would just put more burden on the GC for no gain. But I'm not strongly committed to that stance.

Reply:
Makes sense. No point of caching this when `Class` isn't even caching `getRecordComponents()` or `getPermittedSubclasses()`.

And yes, using classDataAt is easier in bytecode than adding another invokeinterface.